pkg/tfvars/internal/cache/cache.go

```diff
@@ -193,9 +199,90 @@ func DownloadFile(baseURL string, dataType string) (string, error) {
 }
 
 // DownloadImageFile is a helper function that obtains an image file from a given URL,
-// puts it in the cache and returns the local file path.
+// puts it in the cache and returns the local file path.  If the file is compressed
+// by a known compressor, the file is uncompressed prior to being returned.
 func DownloadImageFile(baseURL string) (string, error) {
 	logrus.Infof("Obtaining RHCOS image file from '%v'", baseURL)
 
 	return DownloadFile(baseURL, imageDataType)
 }
+
+func detectContentType(file string) (string, error) {
+	logrus.Debugf("Detecting content type for %s", file)
+	fd, err := os.Open(file)
+	if err != nil {
+		logrus.Errorf("detectContentType unable to open file %s: %v", file, err)
+		return "", err
+	}
+	defer fd.Close()
+
+	// See http://golang.org/pkg/net/http/#DetectContentType for why we use 512
+	buf := make([]byte, 512)
+	_, err = fd.Read(buf)
+	if err != nil {
+		logrus.Errorf("detectContentType unable to read file %s: %v", file, err)
+		return "", err
+	}
+	filetype := http.DetectContentType(buf)
+	logrus.Debugf("detectContentType %s is %s", file, filetype)
+	return filetype, nil
+}
+
+func decompressFileGzip(src, dest string) error {
+	logrus.Debugf("decompressFileGzip %s %s", src, dest)
+	gzipfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	reader, err := gzip.NewReader(gzipfile)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	writer, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+
+	if _, err = io.Copy(writer, reader); err != nil {
+		os.Remove(dest)
+		return err
+	}
+
+	return nil
+}
+
+// decompressFile decompresses data in the cache
+func decompressFile(src, dest string) error {
```

Comment:
this needs a little bit more details around what it is doing in comment.

Reply:
Done.

Reply:
this function is decompressing src to dest, the comment `attempts to decompress a file in the cache` seems not relevant..
`file type is not known, it simply leaves the file untouched` is still moving the file, so untouched in not correct.

Reply:
Obsolete, as we're no longer doing it that way.